### PR TITLE
remove the pulsar admin from pulsar connector

### PR DESCRIPTION
**Pull request by nlu90**
**Please** add a meaningful description for your change here

Connectors should avoid using Pulsar Admin API as much as possible due to permission control issues.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/java/io/pulsar/src/main/java/org/apache/beam/sdk/io/pulsar/PulsarIO.java [hunk 1]
```diff
@@ -87,10 +83,6 @@ abstract Builder setExtractOutputTimestampFn(
       abstract Read build();
     }
 
-    public Read withAdminUrl(String url) {
```
Comment (chamikaramj) [inline sdks/java/io/pulsar/src/main/java/org/apache/beam/sdk/io/pulsar/PulsarIO.java:90]: I don't think we can remove this due to backwards compatibility (seems like this connector not marked experimental).

Can this be addressed by deprecating ?

**Review by chamikaramj — COMMENTED** [review]
Thanks for the contribution.

I'm not that familiar about PulsarIO so not sure if this a change we should pursue. Could you start a Beam dev list discussion to bring this to the attention of folks who have context ?

Also, please add a Github issue with details.

File: sdks/java/io/pulsar/src/main/java/org/apache/beam/sdk/io/pulsar/PulsarIO.java (hunk 1 above)
Comment (lukecwik) [reply sdks/java/io/pulsar/src/main/java/org/apache/beam/sdk/io/pulsar/PulsarIO.java:90]: +1 for deprecating the APIs saying that they are now unused.

File: sdks/java/io/pulsar/src/main/java/org/apache/beam/sdk/io/pulsar/ReadFromPulsarDoFn.java [hunk 2]
```diff
@@ -207,7 +189,7 @@ public OffsetRangeTracker restrictionTracker(
     }
 
     PulsarLatestOffsetEstimator offsetEstimator =
-        new PulsarLatestOffsetEstimator(this.admin, pulsarSource.getTopic());
+        new PulsarLatestOffsetEstimator(null /*this.admin*/, pulsarSource.getTopic());
```
Comment (lukecwik) [inline sdks/java/io/pulsar/src/main/java/org/apache/beam/sdk/io/pulsar/ReadFromPulsarDoFn.java:192]: passing in `null` here will cause an NPE when we try to access the `client` within the `PulsarLatestOffsetEstimator`

**Review by lukecwik — CHANGES_REQUESTED** [review]

**Comment by johnjcasey** [comment]
run java precommit

**Comment by johnjcasey** [comment]
@nlu90 are you still working on Luke's requested changes?

**Comment by johnjcasey** [comment]
Closing as we haven't seen updates

**Comment by johnjcasey** [comment]
If you want this re-opened, just let us know

**Pull request closed without merging**